src/App.js

```diff
@@ -1,44 +1,35 @@
-import PropTypes from 'prop-types';
-import React, { Component, Fragment } from 'react';
-import { withRouter } from 'react-router-dom';
-import { connect } from 'react-redux';
+import React, { Fragment, useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
 import { Routes } from './Routes';
 import { register } from './store';
 import { notifications } from '@redhat-cloud-services/frontend-components-notifications';
 import { NotificationsPortal } from '@redhat-cloud-services/frontend-components-notifications/';
 import './App.scss';
 
-class App extends Component {
+const App = (props) => {
+    const history = useHistory();
+    useEffect(() => {
+        (async () => {
+            const user = await insights.chrome.auth.getUser();
+            if (!user) {
+                location.href = './';
```

Comment:
I would prefer using `history.replace`

Reply:
Yeah, good point! We don't want to allow users to go back to previous page if the user is not defiend.